shotover-proxy/src/frame/cassandra.rs

```diff
@@ -387,27 +388,19 @@ impl CassandraFrame {
         cursor.write_all(&[self.operation.to_opcode().into()]).ok();
 
         serialize_with_length_prefix(&mut cursor, |cursor| {
-            if let Tracing::Response(Some(uuid)) = self.tracing {
-                cursor.write_all(uuid.as_bytes()).ok();
-            }
-
-            if !self.warnings.is_empty() {
-                let warnings_len = self.warnings.len() as i16;
-                cursor.write_all(&warnings_len.to_be_bytes()).ok();
+            if Compression::None == compression {
+                self.write_tracing_and_warnings(cursor);
 
-                for warning in &self.warnings {
-                    let warning_len = warning.len() as i16;
-                    cursor.write_all(&warning_len.to_be_bytes()).ok();
-                    cursor.write_all(warning.as_bytes()).ok();
-                }
-            }
-            if let Compression::None = compression {
-                // Special case None to avoid large copies
                 self.operation.serialize(cursor, self.version)
+
+            // Special case None to avoid large copies
```

Comment:
this looks like the comment is applying to the else block now, we want the comment to apply to the initial if block.